### PR TITLE
Symbol visibility

### PR DESCRIPTION
**Pull request by raffenet**

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/all
test:jenkins/ch4/all

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
@charlesarcher I think this one is finally ready for review. I'm a little baffled by the spawn error tests that failed in Jenkins, but I'm willing to ignore that for the time-being.

**Comment by charlesarcher** [comment]
I'll test this first thing today.

**Comment by charlesarcher** [comment]
This is looking good on my builds.

File: src/mpi/romio/include/mpio.h.in [hunk 1]
```diff
@@ -12,6 +12,12 @@
 
 #include "mpi.h"
 
+#if defined(MPICH_C_HAVE_VISIBILITY)
+#define MPICH_API_PUBLIC __attribute__((visibility ("default")))
```
Comment (raffenet) [inline src/mpi/romio/include/mpio.h.in]: I agree we could probably clean up the macros/defines to be less MPICH-specific. I had meant to do that once I copied to the various sub-components. I'm going to whip up a new version that addresses this. Shouldn't have any functional changes.

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Pull request merged**